File: android/app/src/main/java/ae/findmybay/data/api/VendorApi.kt
package ae.findmybay.data.api

import ae.findmybay.data.api.dto.AvailabilityResponseDto
import ae.findmybay.data.api.dto.VendorDetailDto
import ae.findmybay.data.api.dto.VendorListResponse
import retrofit2.http.GET
import retrofit2.http.Path
import retrofit2.http.Query

interface VendorApi {

    @GET("vendors/nearby")
    suspend fun nearby(
        @Query("lat") lat: Double,
        @Query("lng") lng: Double,
        @Query("radius") radiusMeters: Int = 15_000,
        @Query("serviceId") serviceId: String? = null,
    ): VendorListResponse

    @GET("vendors/{id}")
    suspend fun detail(@Path("id") id: String): VendorDetailDto

    @GET("vendors/{id}/availability")
    suspend fun availability(
        @Path("id") id: String,
        @Query("serviceId") serviceId: String,
        @Query("date") date: String, // YYYY-MM-DD
    ): AvailabilityResponseDto
}

File: ios/FindMyBay/Data/API/VendorAPI.swift
// VendorAPI.swift
//
// Vendor endpoints — parity with `data/api/VendorApi.kt`.

import Foundation

protocol VendorAPI {
    func nearby(lat: Double, lng: Double,
                radiusMeters: Int,
                serviceId: String?) async throws -> VendorListResponse
    func detail(id: String) async throws -> VendorDetailDTO
    func availability(id: String, serviceId: String, date: String) async throws -> AvailabilityResponseDTO
}

struct VendorAPIImpl: VendorAPI {
    let client: APIClient

    func nearby(lat: Double, lng: Double,
                radiusMeters: Int = 15_000,
                serviceId: String? = nil) async throws -> VendorListResponse {
        var query: [URLQueryItem] = [
            .init(name: "lat",    value: String(lat)),
            .init(name: "lng",    value: String(lng)),
            .init(name: "radius", value: String(radiusMeters)),
        ]
        if let serviceId { query.append(.init(name: "serviceId", value: serviceId)) }
        return try await client.request(.GET, path: "vendors/nearby", query: query)
    }

    func detail(id: String) async throws -> VendorDetailDTO {
        try await client.request(.GET, path: "vendors/\(id)")
    }

    func availability(id: String, serviceId: String, date: String) async throws -> AvailabilityResponseDTO {
        try await client.request(
            .GET,
            path: "vendors/\(id)/availability",
            query: [
                .init(name: "serviceId", value: serviceId),
                .init(name: "date",      value: date),
            ]
        )
    }
}
